extensions/devservices/mariadb/src/main/java/io/quarkus/devservices/mariadb/deployment/MariaDBDevServicesProcessor.java

```diff
@@ -39,9 +43,9 @@ public RunningDevServicesDatasource startDatabase(Optional<String> username, Opt
                 QuarkusMariaDBContainer container = new QuarkusMariaDBContainer(imageName, fixedExposedPort,
                         !devServicesSharedNetworkBuildItem.isEmpty());
                 startupTimeout.ifPresent(container::withStartupTimeout);
-                container.withPassword(password.orElse("quarkus"))
-                        .withUsername(username.orElse("quarkus"))
-                        .withDatabaseName(datasourceName.orElse("default"));
+                container.withPassword(password.orElse(DEFAULT_DATABASE_USERNAME))
+                        .withUsername(username.orElse(DEFAULT_DATABASE_PASSWORD))
+                        .withDatabaseName(datasourceName.orElse(DEFAULT_DATABASE_NAME));
```

Comment:
You mixed passwords and usernames up. That's probably why you had issues. Please check all of them and put the username first as it makes more sense.